### PR TITLE
Added rating rubrics page functionality

### PR DESCRIPTION
**Pull request by moezthedev**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes
Video: https://www.loom.com/share/57bbab49b5ad4a0e88add60eb0f05e14

### CODE REVIEW

File: rubrics-ui/prisma/schema.prisma [hunk 1]
```diff
@@ -117,21 +119,26 @@ model Space {
 
 model Program {
   id      String                 @id @default(uuid()) @db.VarChar(64)
-  name    String?                @db.VarChar(64)
-  details String?                @db.VarChar(64)
-  summary String?                @db.VarChar(64)
+  name    String?
```
Comment (RobinNagpal) [inline rubrics-ui/prisma/schema.prisma:122]: This will be mandatory fields

File: rubrics-ui/prisma/schema.prisma [hunk 2]
```diff
@@ -117,21 +119,26 @@ model Space {
...
-  id          String                 @id @default(uuid()) @db.VarChar(64)
-  name        String?                @db.VarChar(64)
-  summary     String?                @db.VarChar(64)
-  description String?                @db.VarChar(64)
-  levels      RubricLevel[]
-  criteria    RubricCriteria[]
-  RubricCell  RubricCell[]
-  programs    ProgramRubricMapping[]
+  id           String                 @id @default(uuid()) @db.VarChar(64)
+  name         String?                @db.VarChar(64)
```
Comment (RobinNagpal) [inline rubrics-ui/prisma/schema.prisma:132]: This will be mandatory fields. At least name and summary
